### DISCUSSION TITLE
About "contributor" definition

### DISCUSSION THREAD

**Discussion by htadashi** [community/community#24300] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I am using GitHub API for a research and got a question about how the list of contributors of a repository is generated. Precisely, supposing that we can get all the <a href="https://docs.github.com/en/developers/webhooks-and-events/github-event-types" rel="noopener nofollow ugc">events</a> that happened in a repository, is the list of its contributors exactly equal to the set of the users that were actors of a <code>PushEvent</code> or a <code>PullRequestEvent</code>?</p>
<p>Thanks for any clarification! :grinning:</p>

**Comment by adlai** [comment, marked as answer]
<p>You may often encounter situations where one individual is the <em>committer</em> of a patch, that was authored by someone else, thus any individual event could add two contributors to that set, if you count both author and committer. You may also want to ignore certain pull request actors, due to the fact that a pull request alone ultimately signifies nothing other than a notification from one repository to another that there is a possibly mergeable patch.</p>

**Comment by htadashi** [comment]
<p>Thanks for the insight, adlai!<br>
Just to confirm, the actor of a pull request appears as a contributor if and only if the pull request is merged to the repository, right? (Assuming that the actor did not made any other commit until then).</p>

**Comment by adlai** [comment]
<aside class="quote no-group" data-username="htadashi" data-post="3" data-topic="158979">
<div class="title">
<div class="quote-controls"></div>
 htadashi:</div>
<blockquote>
<p>if and only if the pull request is merged to the repository</p>
</blockquote>
</aside>
<p>I have no idea. I don’t work for GitHub. Ask the professionals.</p>

**Comment by adlai** [comment]
<aside class="quote no-group" data-username="htadashi" data-post="3" data-topic="158979">
<div class="title">
<div class="quote-controls"></div>
 htadashi:</div>
<blockquote>
<p>the actor of a pull request appears as a contributor</p>
</blockquote>
</aside>
<p>This <a href="https://github.community/t/why-is-my-commit-associated-with-the-wrong-person/10199">other post in the forum</a> has a much more detailed explanation of the various roles.</p>

**Comment by htadashi** [comment]
<p>Thanks for the tip! This post is very helpful :slight_smile:</p>
